src/main.rs

```diff
@@ -89,34 +116,88 @@ fn main() {
                 .takes_value(true)
                 .help("the profile format to output the profiling information in"),
         );
-    let matches = app.get_matches();
-
-    let opt_level: u8 = matches.occurrences_of("optimize").try_into().unwrap();
-    let verbosity_level: u8 = matches.occurrences_of("verbose").try_into().unwrap();
+    app.get_matches()
+}
 
-    // Construct vm:
+/// Create settings by examining command line arguments and environment
+/// variables.
+fn create_settings(matches: &ArgMatches) -> PySettings {
+    let ignore_environment = matches.is_present("ignore-environment");
     let mut settings: PySettings = Default::default();
-    settings.debug = matches.is_present("debug");
-    settings.inspect = matches.is_present("inspect");
-    settings.optimize = opt_level;
-    settings.no_site = matches.is_present("no-site");
-    settings.no_user_site = matches.is_present("no-user-site");
-    settings.verbose = verbosity_level;
-    settings.quiet = matches.is_present("quiet");
-    let vm = VirtualMachine::new(settings);
+    settings.ignore_environment = ignore_environment;
 
-    let res = run_rustpython(&vm, matches);
-    // See if any exception leaked out:
-    handle_exception(&vm, res);
+    if !ignore_environment {
+        settings.path_list.append(&mut get_paths("RUSTPYTHONPATH"));
+        settings.path_list.append(&mut get_paths("PYTHONPATH"));
+    }
 
-    #[cfg(feature = "flame-it")]
+    // Now process command line flags:
+    if matches.is_present("debug") || (!ignore_environment && env::var_os("PYTHONDEBUG").is_some())
     {
-        main_guard.end();
-        if let Err(e) = write_profile(matches) {
-            error!("Error writing profile information: {}", e);
-            process::exit(1);
+        settings.debug = true;
+    }
+
+    if matches.is_present("inspect")
+        || (!ignore_environment && env::var_os("PYTHONINSPECT").is_some())
+    {
+        settings.inspect = true;
+    }
+
+    if matches.is_present("optimize") {
+        settings.optimize = matches.occurrences_of("optimize").try_into().unwrap();
+    } else if !ignore_environment {
+        if let Some(value) = get_env_var_value("PYTHONOPTIMIZE") {
+            settings.optimize = value;
         }
     }
+
+    if matches.is_present("verbose") {
+        settings.verbose = matches.occurrences_of("verbose").try_into().unwrap();
+    } else if !ignore_environment {
+        if let Some(value) = get_env_var_value("PYTHONVERBOSE") {
+            settings.verbose = value;
+        }
+    }
+
+    settings.no_site = matches.is_present("no-site");
+
+    if matches.is_present("no-user-site")
+        || (!ignore_environment && env::var_os("PYTHONNOUSERSITE").is_some())
+    {
+        settings.no_user_site = true;
+    }
+
+    if matches.is_present("quiet") {
+        settings.quiet = true;
+    }
+
+    settings
+}
+
+/// Get environment variable and turn it into integer.
```

Comment:
If you want the env variable to be Unicode anyway, you can just use `env::var`

Reply:
Cool, I'll do it!